cdap-docs/developer-manual/source/metadata/field-lineage.rst

```diff
@@ -0,0 +1,195 @@
+type information. **Field Level Lineage** allows user to get more granular lineage
+view of a dataset. A field lineage for a given dataset shows, for a specified time range, all
+the fields that were computed for a dataset, along with the fields
+from source datasets which participated in computation of those fields. Field lineage also
+shows the detail operations which caused the transformation from fields of a source dataset
+to the field of a given dataset.
+
+|
+
+Data analytics group in an Airline company, wants to perform analytical queries
+on the passenger information stored on SFTP servers. In order to run the workload, passenger
+information is imported to Hadoop. Since the data can come from variety of sources, it is
+normalized into standard format while importing. PII information is also obfuscated. By looking
+at the dataset in Hadoop, data officer (or Business Analyst) wants to understand the meaning of
+the field say "fullName", by reading how it was produced to realize the "fullName" field was
+created by concatenating the "firstName" and "lastName" fields, both of which were extracted as
+positional fields from a CSV record in the source named "passengerList". The Operation team (and
+sometimes a developer wants to identify how a field was computed, typically in a triage/debugging
+scenario where some fields show up with wrong values.
+
+Concepts and Terminology
+========================
+
+- **Field** : Field identifies column in a dataset. Field has a name and data type.
+- **EndPoint** : EndPoint defines the source or destination of the data along with its namespace from where the fields are read or written to.
+- **Field Operation** : Operation defines a single computation on a field. It has a name and description.
+- **Read Operation** : Type of operation that reads from the source EndPoint and creates collection of fields.
+- **Transform Operation** : Type of operation that transforms collection of input fields to collection of output fields.
+- **Write Operation** : Type of operation that write the collection of fields to the destination EndPoint.
+- **Origin** : Origin of the field is the name of the operation which outputted that field. Field can appear in the outputs of multiple operations, which is why ``<origin, fieldName>`` pair is used to uniquely identify the field.
+
+Field Lineage for CDAP Programs
+===============================
+Field Lineage recording is supported from CDAP batch programs - MapReduce and Spark.
+
+|
+
+Consider a simple MapReduce program which is reading the passenger information stored in CSV files
+and normalizes the data into standard format before storing it in the CDAP dataset. Program assumes
+that the fields in the CSV files are - id, firstName, lastName, address. It then concatenates firstName
+and lastName to create fullName. address is normalized into standard format.
+
+|
+
+Following are the field operations that can be recorded by the program in this case -
```

Comment:
Following are the field operations that the program can record in this case:

Reply:
fixed.